src/authentication/team-token.ts

```diff
@@ -0,0 +1,58 @@
+import * as Knex from 'knex';
+import * as moment from 'moment';
+
+const randomstring = require('randomstring');
+
+export const teamTokenLength = 7;
+
+export interface TeamToken {
+  teamId: number;
+  token: string;
+  createdAt: number | moment.Moment;
+}
+export function teamTokenQuery(db: Knex, token?: string, teamId?: number) {
+  const latestTokens = db('teamtoken')
+    .select(db.raw('teamId, MAX(createdAt) as latestStamp'))
+    .groupBy('teamId')
+    .as('latest');
+  const query = db('teamtoken')
+    .join(latestTokens, ((join: any) => join
+      .on('teamtoken.teamId', '=', 'latest.teamId')
+      .andOn('teamtoken.createdAt', '=', 'latest.latestStamp')) as any,
+  );
+  if (token) {
+    if (token.length !== teamTokenLength || !token.match(/^\w+$/)) {
+      throw new Error('Invalid team token');
+    }
+    query.where('teamtoken.token', token);
+  }
+  if (teamId) {
+    if (!teamId.toString().match(/^\d+$/)) {
+      throw new Error('Invalid teamId');
+    }
+    if (token) {
+      query.andWhere('teamtoken.teamId', teamId);
+    } else {
+      query.andWhere('teamtoken.teamId', teamId);
+    }
+  }
+  return query;
+}
+
+export async function validateTeamToken(token: string, db: Knex) {
+  const teamTokens: TeamToken[] = await teamTokenQuery(db, token);
+  if (!teamTokens || teamTokens.length !== 1) {
+    throw new Error('Invalid team token');
```

Comment:
I'm not sure about this since I'm not that familiar with minard-backend or Hapi, but is throwing an error the right way to say that the token is invalid?